feature/amazon/src/main/java/com/revenuecat/purchases/amazon/ProxyAmazonBillingActivityBroadcastReceiver.kt

```diff
@@ -0,0 +1,16 @@
+package com.revenuecat.purchases.amazon
+
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+
+internal class ProxyAmazonBillingActivityBroadcastReceiver(private val activity: Activity) : BroadcastReceiver() {
```

Comment:
Yes, I don't like this... Do you think it would be a better idea to make `ProxyAmazonBillingActivity` implement `BroadcastReceiver`? That way it will finish itself

Reply:
ugh yes, it's a class, I didn't notice. Hmmmm. Yeah I don't think there's any alternative. I am calling unregister in the `onDestroy` and setting it to null so I don't think it will leak, but who knows.

Reply:
Done in [`2cbd49c` (#552)](https://github.com/RevenueCat/purchases-android/pull/552/commits/2cbd49c1c0bc6f40094c75f5e8896fbf7aa2fd5e)